### ISSUE TITLE
Could you please include the makefile?

### ISSUE THREAD

**Issue by LemonCrab666** [hudra0/qosmate#2]
Could you please include the makefile? I'd like to integrate it into my firmware. Thank you for your contribution.

**Comment by hudra0** [comment]
It's on my todo, but I haven't gotten time yet.

**Comment by hudra0** [comment]
I added makefiles for qosmate and the luci app. Please try them and report back if it works for you.

Steps:

1. Navigate to Your OpenWrt Buildroot Directory:
`cd /path/to/your/openwrt
`
2. Clone the QoSmate Package:
`mkdir -p package/qosmate
git clone https://github.com/hudra0/qosmate.git package/qosmate
`
3. Clone the LuCI QoSmate Package:
`mkdir -p package/luci-app-qosmate
git clone https://github.com/hudra0/luci-app-qosmate.git package/luci-app-qosmate
`

**Comment by LemonCrab666** [comment]
> I added makefiles for qosmate and the luci app. Please try them and report back if it works for you.
> 
> Steps:
> 
> 1. Navigate to Your OpenWrt Buildroot Directory:
>    `cd /path/to/your/openwrt `
> 2. Clone the QoSmate Package:
>    `mkdir -p package/qosmate git clone https://github.com/hudra0/qosmate.git package/qosmate `
> 3. Clone the LuCI QoSmate Package:
>    `mkdir -p package/luci-app-qosmate git clone https://github.com/hudra0/luci-app-qosmate.git package/luci-app-qosmate `

Thank you very much for your contribution. I am currently compiling and testing it. As many Chinese users have used your plugin, I have tried to localize Luci in Chinese. Would it be possible to merge it
[qosmate.zh-cn.zip](https://github.com/user-attachments/files/17290012/qosmate.zh-cn.zip)

**Comment by LemonCrab666** [comment]
Restarting or booting the qosmate plugin in OpenWRT will not automatically start, and this file is a log. I am not sure if it is helpful for troubleshooting the problem
[log.txt](https://github.com/user-attachments/files/17484707/log.txt)

**Comment by hudra0** [comment]
> As many Chinese users have used your plugin, I have tried to localize Luci in Chinese. Would it be possible to merge it
[qosmate.zh-cn.zip](https://github.com/user-attachments/files/17290012/qosmate.zh-cn.zip)

Please create a separate issue for this. Multi-language support isn’t quite ready yet, but I’m working on it...

> Restarting or booting the qosmate plugin in OpenWRT will not automatically start, and this file is a log. I am not sure if it is helpful for troubleshooting the problem
[log.txt](https://github.com/user-attachments/files/17484707/log.txt)

Another user has a similar issue. He’s using an unofficial version of OpenWRT. Could that be the case for you as well?

Please show me the output of:

`ubus call system board
`

Could you also run a test for me?

Go to /etc/init.d/qosmate and remove the following function:
```
service_triggers() {
    procd_add_reload_trigger "network"
    procd_add_reload_trigger "firewall"
}

```

Save the file.

Restart qosmate with `service qosmate restart`.

Reboot the router and wait 2-3 minutes.

Check if qosmate is running (not via UI) by entering: `service qosmate status`.

If it’s not running, show me the logs and the output of the status.

**Comment by hudra0** [comment]
Probably fixed with 1d2936883604bda0d359204a5a8fa5c7aad81081
